beets/autotag/match.py

```diff
@@ -131,11 +131,15 @@ def assign_items(
     costs = [[float(track_distance(i, t)) for t in tracks] for i in items]
     # Find a minimum-cost bipartite matching.
     log.debug("Computing track assignment...")
-    cost, _, assigned_idxs = lap.lapjv(np.array(costs), extend_cost=True)
+    cost, _, assigned_item_idxs = lap.lapjv(np.array(costs), extend_cost=True)
     log.debug("...done.")
 
     # Produce the output matching.
-    mapping = {items[i]: tracks[t] for (t, i) in enumerate(assigned_idxs)}
+    mapping = {
+        items[iidx]: tracks[tidx]
+        for tidx, iidx in enumerate(assigned_item_idxs)
+        if iidx != -1
```

Comment:
Maybe, to be extra-safe against this type of issue (which seems to be pretty subtle, at a first glance I was very confused why the testcase didn't catch it in the first place), the comparison could be turned into
```suggestion
        if iidx >= 0
```


This might also be worth a code comment about the meaning of `-1`? At a first glance, the `lapvj` documentation doesn't mention it.

Reply:
I added some comments in the logic and tried making it a bit cleaner. Do you prefer `iidx >= 0` over `iidx != -1`? I find the latter more explicit and self-documenting regarding the expected index for unmatched tracks.

Reply:
Sounds good! Let's stick to the current version: I guess it's exceedingly unlikely that lapjv could ever start returning other negative values. (And either way, our tests should hopefully catch that case now.)